src/mongocxx/write_concern.cpp

```diff
@@ -96,9 +107,25 @@ stdx::optional<std::int32_t> write_concern::nodes() const {
 }
 
 stdx::optional<write_concern::level> write_concern::acknowledge_level() const {
+    stdx::optional<write_concern::level> ack_level;
     std::int32_t w = libmongoc::write_concern_get_w(_impl->write_concern_t);
-    return w < 1 ? stdx::optional<write_concern::level>{static_cast<write_concern::level>(w)}
-                 : stdx::nullopt;
+    switch (w) {
+    case 0:
+        ack_level = write_concern::level::k_unacknowledged;
+        break;
+    case -2:
+        ack_level = write_concern::level::k_default;
+        break;
+    case -3:
+        ack_level = write_concern::level::k_majority;
+        break;
+    case -4:
+        ack_level = write_concern::level::k_tag;
+        break;
+    default:
+        ack_level = write_concern::level::k_unknown;
+    }
+    return w < 1 ? ack_level : stdx::nullopt;
```

Comment:
My other comment obviates this by eliminating the ternary op, but better to always parenthesize:

return (w < 1) ? ...